anta/tests/avt.py

```diff
@@ -0,0 +1,232 @@
+
+from pydantic import BaseModel
+
+from anta.models import AntaCommand, AntaTemplate, AntaTest
+from anta.tools import get_value
+
+
+class VerifyAVTPathHealth(AntaTest):
+    """
+    Verifies the status of all Adaptive Virtual Topology (AVT) paths for all VRFs.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all AVT paths for all VRFs are active and valid.
+    * Failure: The test will fail if the AVT path is not configured or if any AVT path under any VRF is either inactive or invalid.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.avt:
+      - VerifyAVTPathHealth:
+    ```
+    """
+
+    name = "VerifyAVTPathHealth"
+    description = "Verifies the status of all AVT paths for all VRFs."
+    categories: ClassVar[list[str]] = ["avt"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show adaptive-virtual-topology path")]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyAVTPathHealth."""
+        # Initialize the test result as success
+        self.result.is_success()
+
+        # Get the command output
+        command_output = self.instance_commands[0].json_output.get("vrfs", {})
+
+        # Check if AVT is configured
+        if not command_output:
+            self.result.is_failure("Adaptive virtual topology paths are not configured.")
+            return
+
+        # Iterate over each VRF
+        for vrf, vrf_data in command_output.items():
+            # Iterate over each AVT path
+            for profile, avt_path in vrf_data.get("avts", {}).items():
+                for path, flags in avt_path.get("avtPaths", {}).items():
+                    # Get the status of the AVT path
+                    valid = flags["flags"]["valid"]
+                    active = flags["flags"]["active"]
+
+                    # Check the status of the AVT path
+                    if not valid and not active:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is invalid and not active.")
+                    elif not valid:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is invalid.")
+                    elif not active:
+                        self.result.is_failure(f"AVT path {path} for profile {profile} in VRF {vrf} is not active.")
+
+
+class VerifyAVTSpecificPath(AntaTest):
+    """
+    Verifies the status and type of an Adaptive Virtual Topology (AVT) path for a specified VRF.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the AVT path is active, valid, and matches the specified type (direct/multihop) for the given VRF.
+               If no path type is specified, both direct and multihop paths are considered.
+    * Failure: The test will fail if the AVT path is not configured or if the AVT path is not active, valid, or does not match the specified type for the given VRF.
```

Comment:
Maybe this need to be reworded a bit? Looks like it is not mentioning that we would configured multiple paths if there are multiple in the Succes criteria

Reply:
@gmuloc Is this looks ok?
```
Expected Results
    ----------------
    * Success: The test will pass if all AVT paths for the specified VRF are active, valid, and match the specified type (direct/multihop) if provided.
               If multiple paths are configured, the test will pass as long as at least one path meets the specified criteria.
    * Failure: The test will fail if no AVT paths are configured for the specified VRF, or if any configured path is not active, valid, or does not match the specified type.
```

Reply:
`the test will pass as long as at least one path meets the specified criteria.`

I don't think this is what the test does - to succeed you will need all path to meet right? Given that you will go over all path. (which I think is probably good as we would want to know if any is failing - though later on we could add some flag to say "all/any" as a success criteria ;)